src/Controller/Component/PrgComponent.php

```diff
@@ -2,26 +2,70 @@
 namespace Search\Controller\Component;
 
 use Cake\Controller\Component;
-use Cake\Event\Event;
 
 class PrgComponent extends Component
 {
+
+    /**
+     * Default config
```

Comment:
This doc block could do with some explanation in my opinion about the `actions` configuration setting. Due to it possibly containing `bool`, `string` or `array`. From reading the code I assume you configure which controller actions you want the PrgComponent to run on, in which case wouldn't `index` be the default as it's the most used use-case.